docarray/utils/lookup.py

```diff
@@ -0,0 +1,325 @@
+"""
+
+Originally from https://github.com/naiquevin/lookupy
+
+The library is provided as-is under the MIT License
+
+Copyright (c) 2013 Vineet Naik (naikvin@gmail.com)
+
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+import re
+from typing import TYPE_CHECKING, List, Union, Any, Sequence
+
+from functools import partial
+
+if TYPE_CHECKING:  # pragma: no cover
+    from docarray import BaseDocument
+
+PLACEHOLDER_PATTERN = re.compile(r'\{\s*([a-zA-Z0-9_]*)\s*}')
+
+
+def dunder_get(_dict: Any, key: str) -> Any:
```

Comment:
I think we have to unify this, if i am not wrong for nested access we use `a.b` instead of `a__b`. Personally i prefer the dunder bc it is more string-y, but it should be the same everywhere

Reply:
I have checked, and MongoDB does `.` access, and since the QueryLanguage is given by MongoDB, I would change